pkg/authorization/client/clientset_generated/internalclientset/clientset.go

```diff
@@ -0,0 +1,73 @@
+package internalclientset
+
+import (
+	"github.com/golang/glog"
+	v1core "github.com/openshift/origin/pkg/authorization/client/clientset_generated/internalclientset/typed/core/v1"
+	restclient "k8s.io/kubernetes/pkg/client/restclient"
+	discovery "k8s.io/kubernetes/pkg/client/typed/discovery"
+	"k8s.io/kubernetes/pkg/util/flowcontrol"
+)
+
+type Interface interface {
+	Discovery() discovery.DiscoveryInterface
```

Comment:
I don't think this work, because https://github.com/kubernetes/kubernetes/blob/cbab337b9967663f581165da18126f660b7fa7a8/pkg/client/typed/discovery/discovery_client.go#L291

I think we will have to make the `LegacyPrefix` configurable?